client/src/main/java/io/split/storages/memory/InMemoryCacheImp.java

```diff
@@ -160,4 +183,29 @@ public Set<String> getSegments() {
         return _concurrentMap.values().stream()
                 .flatMap(parsedSplit -> parsedSplit.getSegmentsNames().stream()).collect(Collectors.toSet());
     }
-}
+
+    private void addToFlagSets(ParsedSplit featureFlag) {
+        HashSet<String> sets = featureFlag.flagSets();
+        if( sets != null) {
+            for (String set: sets) {
+                if (!_flagSetsFilter.Intersect(set)) {
+                    continue;
+                }
+                HashSet<String> features = _flagSets.get(set);
+                if (features == null) {
+                    features = new HashSet<>();
+                }
+                features.add(featureFlag.feature());
+                _flagSets.put(set, features);
+            }
+        }
+    }
+
+    private void removeFromFlagSets(String featureFlagName, HashSet<String> sets) {
```

Comment:
can `sets` be null? check that case if necessary.

Reply:
It's true. So now we validate if not null, and I updated the test cases.